conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt

```diff
@@ -0,0 +1,515 @@
+
+        private val olToPostgres = mapOf<Permission, Set<PostgresPrivileges>>(
+                Permission.READ to EnumSet.of(PostgresPrivileges.SELECT),
+                Permission.WRITE to EnumSet.of(PostgresPrivileges.INSERT, PostgresPrivileges.UPDATE),
+                Permission.OWNER to EnumSet.of(PostgresPrivileges.ALL)
+        )
+
+        private val allViewPermissions = setOf(Permission.READ)
+
+        private val allTablePermissions = setOf(Permission.READ, Permission.WRITE, Permission.OWNER)
+
+        private fun fqnToColumnName(fqn: FullQualifiedName): String {
+            return fqn.toString()
+        }
+    }
+
+    internal fun configureRolesInDatabase(dataSource: HikariDataSource) {
+        val roles = principalsMapManager.getAllRoles()
+
+        if (roles.isNotEmpty()) {
+            val roleIds = roles.map { PostgresRoles.buildPostgresRoleName(it.id) }
+            val roleIdsSql = roleIds.joinToString { DataTables.quote(it) }
+
+            dataSource.connection.use { connection ->
+                connection.createStatement().use { statement ->
+
+                    logger.info("Revoking ${Schemas.PUBLIC_SCHEMA} schema right from roles: {}", roleIds)
+                    //Don't allow users to access public schema which will contain foreign data wrapper tables.
+                    statement.execute("REVOKE USAGE ON SCHEMA ${Schemas.PUBLIC_SCHEMA} FROM $roleIdsSql")
+                }
+            }
+        }
+    }
+
+    override fun createRole(role: Role) {
+        val (dbRole, _) = dbCredentialService.getOrCreateRoleAccount(role)
+
+        logger.debug("Creating role if not exists {}", dbRole)
+        atlas.connection.use { connection ->
+            connection.createStatement().use { statement ->
+                statement.execute(createRoleIfNotExistsSql(dbRole))
+                //Don't allow users to access public schema which will contain foreign data wrapper tables.
+                logger.debug("Revoking ${Schemas.PUBLIC_SCHEMA} schema right from role: {}", role)
+                statement.execute("REVOKE USAGE ON SCHEMA ${Schemas.PUBLIC_SCHEMA} FROM ${DataTables.quote(dbRole)}")
+            }
+        }
+    }
+
+    override fun createUnprivilegedUser(principal: SecurablePrincipal) {
+        /**
+         * To simplify work-around for ESRI username limitations, we are only introducing one additional
+         * field into the dbcreds table. We keep the results of calling [buildPostgresUsername] as the lookup
+         * key, but instead use the username and password returned from the db credential service.
+         */
+        val (dbUser, dbUserPassword) = dbCredentialService.getOrCreateDbAccount(principal)
+
+        logger.debug("Creating user if not exists {}", dbUser)
+        atlas.connection.use { connection ->
+            connection.createStatement().use { statement ->
+                //TODO: Go through every database and for old users clean them out.
```

Comment:
this is a super old todo and not actually important for this pr, yeah?

Reply:
Yep, just ported straight from ACM